src/components/team-filter-header.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+<ion-header>
+    <ion-toolbar>
+      <ion-segment scrollable @click="teamSelected($event)"  v-if ="newTeamArray.length==0">
+         <ion-segment-button value="all">
+          <ion-label>All</ion-label>
+        </ion-segment-button>
+        <ion-segment-button  v-for="item in teamName" :key="item.objectId" :value="item.name">
+          <ion-label>{{item.name}}</ion-label>
+        </ion-segment-button>
+        <ion-segment-button value="setting">
+          <ion-icon :icon="settingIcon"></ion-icon>
+        </ion-segment-button>
+      </ion-segment>
+
+      <!-- When setting data set -->
+      <ion-segment scrollable @click="teamSelected($event)"  v-if ="newTeamArray.length!=0">
+          <ion-segment-button value="all">
+          <ion-label>All</ion-label>
+        </ion-segment-button>
+
+        <ion-list v-for="(item, index) in newTeamArray" v-bind:index="item.priority" :key="index">
+        <ion-segment-button v-if="item.toggle" :value="item.name">
+          <div class="segment-block">
+          <ion-avatar v-if="item.isIcon && item.icon!==''" size="1.5rem" slot="start">
+                <img v-bind:src="item.icon" />
+            </ion-avatar>
+            <ion-label class="ion-margin-start">{{item.name}}</ion-label>
+          </div>
+        </ion-segment-button>
+        </ion-list>
+
+        <ion-segment-button value="setting">
+          <ion-icon :icon="settingIcon"></ion-icon>
+        </ion-segment-button>
+      </ion-segment>
```

Comment:
this shouldn't need to know about the different states and ways the settings object is organised, but rather just use one or a few `getter`s to create the segments. Also, if I am reading this right, this appears even if we are only member of only team (which case, all and that team are the same so we don't want to show the segment at all).

Together with the changes below, that would reduce the entire thing here to something about this... (untested - misses the icon thing right now):

```suggestion
      <ion-segment scrollable @click="teamSelected($event)"  v-if ="myTeams.length > 1">
        <ion-segment-button  v-for="item in teamListing" :key="item.value" :value="item.value">
          <ion-label>{{item.name}}</ion-label>
        </ion-segment-button>
        <ion-segment-button value="setting">
          <ion-icon :icon="settingIcon"></ion-icon>
        </ion-segment-button>
      </ion-segment>
```